vmc/resource_vmc_site_recovery.go

```diff
@@ -0,0 +1,205 @@
+	connector := (m.(*ConnectorWrapper)).Connector
+	sddcID := d.Id()
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	siteRecoveryClient := draas.NewDefaultSiteRecoveryClient(connector)
+	siteRecovery, err := siteRecoveryClient.Get(orgID, sddcID)
+	if err != nil {
+		if err.Error() == errors.NewNotFound().Error() {
+			log.Printf("Site recovery for SDDC with ID %s not found", sddcID)
+			d.SetId("")
+			return fmt.Errorf("Site recovery for SDDC with ID %s not found", sddcID)
+		}
+		return fmt.Errorf("Error while site recovery information for SDDC with ID %s : %v", sddcID, err)
+	}
+	d.SetId(siteRecovery.Id)
+	d.Set("site_recovery_state", siteRecovery.SiteRecoveryState)
+	d.Set("draas_h5_url", siteRecovery.DraasH5Url)
+	d.Set("user_id", siteRecovery.UserId)
+	d.Set("user_name", siteRecovery.UserName)
+
+	srm_nodes := []map[string]string{}
+	for _, srmNode := range siteRecovery.SrmNodes {
+		m := map[string]string{}
+		m["id"] = *srmNode.Id
+		m["ip_address"] = *srmNode.IpAddress
+		m["host_name"] = *srmNode.Hostname
+		m["state"] = *srmNode.State
+		m["type"] = *srmNode.Type_
+		m["vm_moref_id"] = *srmNode.VmMorefId
+		srm_nodes = append(srm_nodes, m)
+	}
+
+	vr_node := map[string]string{}
+	vr_node["vm_moref_id"] = *siteRecovery.VrNode.VmMorefId
+	vr_node["id"] = *siteRecovery.VrNode.Id
+	vr_node["hostname"] = *siteRecovery.VrNode.Hostname
+	vr_node["type"] = *siteRecovery.VrNode.Type_
+	vr_node["state"] = *siteRecovery.VrNode.State
+	vr_node["ip_address"] = *siteRecovery.VrNode.IpAddress
+
+	d.Set("srm_nodes", srm_nodes)
+	d.Set("vr_node", vr_node)
+	return nil
+}
+
+func resourceSiteRecoveryDelete(d *schema.ResourceData, m interface{}) error {
+	connector := (m.(*ConnectorWrapper)).Connector
+	siteRecoveryClient := draas.NewDefaultSiteRecoveryClient(connector)
+
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	sddcID := d.Get("sddc_id").(string)
+
+	task, err := siteRecoveryClient.Delete(orgID, sddcID, nil, nil)
+	if err != nil {
+		return fmt.Errorf("Error while deactivating site recovery for sddc %s: %v", sddcID, err)
+	}
+	tasksClient := draas.NewDefaultTaskClient(connector)
+	return resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
+		task, err := tasksClient.Get(orgID, task.Id)
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Error while deactivating site recovery for sddc %s : %v", sddcID, err))
+		}
+		if *task.Status != "FINISHED" {
+			return resource.RetryableError(fmt.Errorf("Expected instance to be deleted but was in state %s", *task.Status))
+		}
+		d.SetId("")
+		return resource.NonRetryableError(nil)
+	})
+}
+
+func resourceSiteRecoveryUpdate(d *schema.ResourceData, m interface{}) error {
+	if d.HasChange("srm_extension_key_suffix") {
+		err := resource.NonRetryableError(resourceSiteRecoveryDelete(d, m))
+		if err != nil {
+			return fmt.Errorf("Error while deactivating site recovery: %v", err)
+		}
+		time.Sleep(15 * time.Minute)
```

Comment:
can you add a comment regarding the sleep?

Reply:
Done